lhotse/cut.py

```diff
@@ -1123,6 +1124,41 @@ def split(self, num_splits: int, randomize: bool = False) -> List['CutSet']:
             split_sequence(self, num_splits=num_splits, randomize=randomize)
         ]
 
+    def subset(self, supervision_ids: Optional[Iterable[str]] = None) -> 'CutSet':
+        """
+        Return a new CutSet with Cuts containing only `supervision_ids`.
+
+        Example:
+            >>> cuts = CutSet.from_yaml('path/to/cuts')
+            >>> train_set = cuts.subset(supervision_ids=train_ids)
+            >>> test_set = cuts.subset(supervision_ids=test_ids)
+
+        :param supervision_ids: List of `supervision_ids` to keep
+        :return: a CutSet with filtered supervisions
+        """
+
+        assert supervision_ids is not None, \
+            "supervision ids can not be None."
+
+        # use set for faster search
+        supervision_ids = set(supervision_ids)
+
+        # fails test if fastcopy is used
+        filtered_cutset = deepcopy(self)
```

Comment:
I think this can get pretty slow for large `CutSet`s; I'd suggest the following approach instead:

```
filtered_cutset = CutSet.from_cuts(
    cut for cut in self if any(s.id in supervision_ids for s in cut.supervisions)
)
```

BTW do you think that when a cut contains more than one supervision, and only one is in the supplied list, we should drop the supervisions outside of it? That would be more complex to handle as `MixedCut.supervisions` cannot be set directly (as in line 1150 below).

Reply:
(I think to handle that properly we could add `.filter_supervisions` methods to each of the three `Cut` classes that returns a copy of the cut - maybe using `fastcopy` - with filtered supervisions; that'd be consistent with the `map_supervisions` method that I think is already there)

Reply:
Thanks, ok I will replace deepcopy by comprehension. :) regarding dropping supervisions not mentioned in supervision ids, I think dropping them entirely makes sense. filter_supervisions seems like a good solution for this given that Mixed cuts supervisions cannot be reassigned easily. I will look into it

Reply:
@pzelasko FYI I added `filter_supervisions` to all Cut variants and adjusted the subset method according to your comment.